test/unit-test/mqtt_agent_utest.c

```diff
@@ -853,3 +909,194 @@ void test_MQTTAgent_Terminate_success( void )
     TEST_ASSERT_NULL( command.pArgs );
     TEST_ASSERT_EQUAL_PTR( stubCompletionCallback, command.pCommandCompleteCallback );
 }
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior when sent command is NULL.
+ */
+void test_MQTTAgent_CommandLoop_sent_command_null( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTAgentCommandFuncReturns_t pReturnFlags;
+
+    setupAgentContext( &mqttAgentContext );
+
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    pReturnFlags.addAcknowledgment = false;
+    pReturnFlags.runProcessLoop = true;
+    pReturnFlags.endLoop = true;
+
+    MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_ProcessLoop_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTT_ProcessLoop_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior when sent command is not NULL.
+ */
+void test_MQTTAgent_CommandLoop_with_sent_command( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTAgentCommandFuncReturns_t pReturnFlags;
+    Command_t command_to_send;
+
+    setupAgentContext( &mqttAgentContext );
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    pReturnFlags.addAcknowledgment = false;
+    pReturnFlags.runProcessLoop = true;
+    pReturnFlags.endLoop = true;
+
+    MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTT_ProcessLoop_IgnoreAndReturn( MQTTSuccess );
+
+    /* Initializing command to be sent to the commandLoop. */
+    command_to_send.commandType = PUBLISH;
+    command_to_send.pCommandCompleteCallback = stubCompletionCallback;
+    command_to_send.pCmdContext = NULL;
+    command_to_send.pArgs = NULL;
+    mqttAgentContext.agentInterface.pMsgCtx->pSentCommand = &command_to_send;
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Making Sure that callback is called. */
+    TEST_ASSERT_EQUAL( 1, commandCompleteCallbackCount );
+}
+
+/**
+ * @brief Test that MQTTAgent_CommandLoop adds acknowledgment in pendingAcks for the commands requiring
+ * acknowledgments.
+ */
+void test_MQTTAgent_CommandLoop_add_acknowledgment_success( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    MQTTAgentCommandFuncReturns_t pReturnFlags;
+    Command_t command_to_send;
```

Comment:
This should be lower camel case
```suggestion
    Command_t commandToSend;
```